src/ResourceManager/OperationalInsights/Commands.OperationalInsights/Query/InvokeOperationalInsightsQuery.cs

```diff
@@ -0,0 +1,100 @@
+﻿using Microsoft.Azure.Commands.Common.Authentication;
+using Microsoft.Azure.Commands.OperationalInsights.Models;
+using Microsoft.Azure.OperationalInsights;
+using Microsoft.Rest;
+using System;
+using System.Management.Automation;
+using System.Net;
+
+namespace Microsoft.Azure.Commands.OperationalInsights.Query
+{
+    [Cmdlet("Invoke", "AzureRmOperationalInsightsQuery"), OutputType(typeof(PSQueryResponse))]
+    public class InvokeOperationalInsightsQuery : ResourceManager.Common.AzureRMCmdlet
+    {
+        [Parameter(Mandatory = true, HelpMessage = "The workspace ID.")]
+        public string WorkspaceId { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The query to execute.")]
+        public string Query { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "The timespan to bound the query by.")]
+        public TimeSpan? Timespan { get; set; } = null;
+
+        [Parameter(Mandatory = false, HelpMessage = "Puts an upper bound on the amount of time the server will spend processing the query. See: https://dev.loganalytics.io/documentation/Using-the-API/Timeouts")]
+        [ValidateRange(1, int.MaxValue)]
+        public int Wait { get; set; } = int.MinValue;
+
+        [Parameter(Mandatory = false, HelpMessage = "If specified, rendering information for metric queries will be included in the response.")]
+        public SwitchParameter IncludeRender { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "If specified, query statistics will be included in the response.")]
+        public SwitchParameter IncludeStatistics { get; set; }
+
+        [Parameter(Mandatory = false,
+            HelpMessage = "When specified, uses the 'DEMO_KEY' API key for authentication. This is only valid when querying against the workspace 'DEMO_WORKSPACE'.")]
+        public SwitchParameter UseDemoKey { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "For internal use only.")]
```

Comment:
Why do you have parameters for internal use only? Why would you want the user to be able to see these parameters and why do you need them?

Reply:
This wasn't desirable. I was continuing to search for a way to resolve this and settled on introducing another field in the AzureEnvironment. Originally this was to support testing against INT, but then I realized that having it in AzureEnvironment was the correct long term approach for national clouds.